__tests__/integrations/Braze/Braze.test.js

```diff
@@ -0,0 +1,356 @@
+    expect(window.braze.getUser().setDateOfBirth).toHaveBeenCalledWith(1990, 1, 1);
+    // Expect any other necessary Braze methods to be called
+  });
+
+  it('should handle supportDedup and update user attributes when traits have changed', () => {
+    jest.clearAllMocks();
+    const config = {
+      appKey: 'APP_KEY',
+      trackAnonymousUser: true,
+      enableBrazeLogging: false,
+      supportDedup: true,
+      dataCenter: 'US-03',
+      enableHtmlInAppMessages: false,
+      allowUserSuppliedJavascript: false,
+    };
+    const analytics = {};
+    const destinationInfo = {};
+
+    const braze = new Braze(config, analytics, destinationInfo);
+    braze.init();
+    // mock the window.braze
+    window.braze = {
+      initialize: jest.fn(),
+      automaticallyShowInAppMessages: jest.fn(),
+      changeUser: jest.fn(),
+      openSession: jest.fn(),
+      getUser: jest.fn().mockReturnThis(),
+      setCountry: jest.fn(),
+      setHomeCity: jest.fn(),
+      setDateOfBirth: jest.fn(),
+      setEmail: jest.fn(),
+      setFirstName: jest.fn(),
+      setGender: jest.fn(),
+      setLastName: jest.fn(),
+      setPhoneNumber: jest.fn(),
+      setCustomUserAttribute: jest.fn(),
+      logCustomEvent: jest.fn(),
+      getCachedContentCards: jest.fn(),
+      getCachedFeed: jest.fn(),
+      User: {
+        Genders: {
+          OTHER: 'o',
+          MALE: 'm',
+          FEMALE: 'f',
+        },
+      },
+    };
+
+    // Create a mock rudderElement with necessary properties
+    const rudderElement = {
+      message: {
+        userId: 'user123',
+        context: {
+          traits: {
+            email: 'updated@example.com',
+            firstName: 'David',
+            lastName: 'Doe',
+            gender: 'female',
+            phone: '1234567890',
+            address: {
+              country: 'USA',
+              city: 'Austin',
+            },
+            birthday: '1990-01-01',
+          },
+        },
+      },
+    };
+
+    // Mock previous payload in Storage
+    Storage.getItem.mockReturnValueOnce({
+      userId: 'user123',
+      context: {
+        traits: {
+          email: 'test@example.com',
+          firstName: 'John',
+          lastName: 'Doe',
+          gender: 'male',
+          phone: '1234567890',
+          address: {
+            country: 'USA',
+            city: 'New York',
+          },
+          birthday: '1990-01-01',
+        },
+      },
+    });
+
+    jest.spyOn(window.braze, 'changeUser');
+    jest.spyOn(window.braze.getUser(), 'setEmail');
+    jest.spyOn(window.braze.getUser(), 'setFirstName');
+    jest.spyOn(window.braze.getUser(), 'setLastName');
+    jest.spyOn(window.braze.getUser(), 'setGender');
+    jest.spyOn(window.braze.getUser(), 'setPhoneNumber');
+    jest.spyOn(window.braze.getUser(), 'setCountry');
+    jest.spyOn(window.braze.getUser(), 'setHomeCity');
+    jest.spyOn(window.braze.getUser(), 'setDateOfBirth');
```

Comment:
can this be moved to function

Reply:
Once we mock the window which we are doing in each jest block then only we can spy on else it will not be able to spy on the object